examples/datasets/load_limo_dataset.py

```diff
@@ -0,0 +1,173 @@
+
+The code allows to:
+
+    - Fetch single subjects epochs data for the LIMO data set.
+    - Get the epochs information from the LIMO .mat files, such as
+      sampling rate, number of epochs per condition, number and name
+      of EEG channels per subject.
+    - Create a custom info and epochs structure in MNE-Python.
+    - Interpolate missing channels.
+    - Plot the evoked response for experimental conditions.
+
+.. note:: This example assumes that the LIMO data set has already
+          been downloaded and stored in home directory.
+
+
+References
+----------
+.. [1] Guillaume, Rousselet. (2016). LIMO EEG Dataset, [dataset].
+       University of Edinburgh, Centre for Clinical Brain Sciences.
+       https://doi.org/10.7488/ds/1556.
+"""
+
+# Authors: Jose C. Garcia Alanis <alanis.jcg@gmail.com>
+#
+#
+# License: BSD (3-clause)
+
+
+# import necessary packages
+import os.path as op
+import scipy.io
+import numpy as np
+import mne
+
+print(__doc__)
+
+
+# --- define load limo epochs function ---
+
+
+def load_limo_epochs(subject, path=None, interpolate=False):
+    """Fetch subjects epochs data for the LIMO data set.
+    Parameters
+    ----------
+    path : str
+        The path to store the downloaded data. Defaults to home directory.
+    subject : int | str
+        Subject to use. Can be in the range from 2 to 18.
+        If string, must be 'S1', 'S2' etc.
+    interpolate : bool
+        Whether to interpolate missing channels.
+    Returns
+    -------
+    epochs : MNE Epochs data structure
+        The epochs.
+    """
+
+    # set path to data
+    if path is None:
+        limo_path = op.expanduser('HOME/limo_dataset')
+        if not op.exists(limo_path):
+            raise ValueError('Could not find path to limo data set.')
+    else:
+        if isinstance(path, str):
+            limo_path = path
+        else:
+            raise ValueError('"path" argument must be string or None.')
+
+    # subject in question
+    if isinstance(subject, int):
+        sub = 'S%i' % subject
+    elif isinstance(subject, str):
+        if not subject.startswith('S'):
+            raise ValueError('`subject` must start with `S`')
+        sub = subject
+
+    # -- 1) import .mat files
+    # epochs info
+    fname_info = op.join(limo_path, sub, 'LIMO.mat')
+    data_info = scipy.io.loadmat(fname_info)
+    # epochs data
+    fname_eeg = op.join(limo_path, sub, 'Yr.mat')
+    data = scipy.io.loadmat(fname_eeg)
+
+    # -- 2) get epochs information from structure
+    # sampling rate
+    sfreq = data_info['LIMO']['data'][0][0][0][0]['sampling_rate'][0][0]
+    # tmin and tmax
+    tmin = data_info['LIMO']['data'][0][0][0][0]['start'][0][0]
+    # number of epochs per condition
+    design = data_info['LIMO']['design'][0][0]['X'][0][0]
+
+    # create events matrix
+    events = np.zeros(shape=(len(design), 3), dtype=int)
+    events[:, 0] = list(range(len(design)))
+    events[:, 2] = design[:, 1]
```

Comment:
It might actually be more economic to just construct it properly in the first place:

`np.array([list(range(len(design))), np.zeros(len(design)), design[:, 1]])`

Or something like that.